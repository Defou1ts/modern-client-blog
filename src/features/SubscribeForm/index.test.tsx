import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import { SubscribeForm } from './index';

describe('SubscribeForm', () => {
	it('renders a  SubscribeForm', () => {
		render(<SubscribeForm />);
		const text = screen.getByText('subscribe.title');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('SubscribeForm snapshot', () => {
		const component = renderer.create(<SubscribeForm />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
