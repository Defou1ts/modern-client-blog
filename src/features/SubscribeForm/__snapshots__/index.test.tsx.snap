// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeForm SubscribeForm snapshot 1`] = `
<div
  className="formWrapper"
>
  <h2
    className="h h2 title"
  >
    subscribe.title
  </h2>
  <form
    action="#"
    className="form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="wrapper"
    >
      <input
        className="input input"
        data-test-id="subsribe-form-input"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="subscribe.placeholder"
        value=""
      />
    </div>
    <button
      className="button primary"
      data-test-id="subcribe-form-submit"
      type="submit"
    >
      subscribe.button
    </button>
  </form>
</div>
`;
